Teacher/Teacher/SpeedGrader/MainLayout/View/SubmissionHeaderView.swift

```diff
@@ -31,40 +31,58 @@ struct SubmissionHeaderView: View {
     var groupName: String? { isGroupSubmission ? submission.groupName : nil }
     var routeToSubmitter: String? {
         if isGroupSubmission {
-            return nil
+            nil
         } else {
-            return "/courses/\(assignment.courseID)/users/\(submission.userID)"
+            "/courses/\(assignment.courseID)/users/\(submission.userID)"
         }
     }
 
     var body: some View {
         HStack(spacing: 0) {
-            Button(action: navigateToSubmitter, label: {
-                HStack(spacing: 0) {
+            Button(action: navigateToSubmitter) {
+                HStack(spacing: 12) {
                     avatar
 
                     VStack(alignment: .leading, spacing: 2) {
                         nameText
-                            .font(.semibold16).foregroundColor(.textDarkest)
-                        HStack(spacing: 2) {
-                            Image(uiImage: submission.status.icon)
-                                .size(uiScale.iconScale * 18)
-                                .foregroundStyle(Color(submission.status.color))
-                            Text(submission.status.text)
-                                .font(.medium14).foregroundColor(Color(submission.status.color))
+                            .font(.semibold16)
+                            .foregroundStyle(.textDarkest)
+
+                        HStack(spacing: 4) {
+                            HStack(spacing: 2) {
+                                Image(uiImage: submission.status.icon)
+                                    .size(uiScale.iconScale * 16)
+                                    .foregroundStyle(Color(submission.status.color))
+
+                                Text(submission.status.text)
+                                    .font(.regular14)
+                                    .foregroundStyle(Color(submission.status.color))
+                            }
+
+                            Color.borderMedium
+                                .frame(width: 1)
+                                .clipShape(RoundedRectangle(cornerRadius: 2))
+
+                            Text(assignment.dueText)
+                                .font(.regular14)
+                                .foregroundStyle(.textDark)
                         }
+                        .fixedSize(horizontal: false, vertical: true)
                     }
-                        .padding(.leading, 12)
                 }
-                    .padding(EdgeInsets(top: 16, leading: 16, bottom: 8, trailing: 0))
-            })
-                .buttonStyle(PlainButtonStyle())
-                .identifier("SpeedGrader.userButton")
+                .padding(.leading, 6)
+                .padding(.trailing, 16)
+            }
+            .padding(.init(top: 12, leading: 16, bottom: 12, trailing: 0))
```

Comment:
```suggestion
                .paddingStyle(.leading, .cellIconLeading)
                .paddingStyle(.trailing, .standard)
                .padding(.vertical, 8)
            }
```
To match figma and to extend touch area to left/top/bottom